### ISSUE TITLE
Pinned chats

### ISSUE THREAD

**Issue by stek29** [zhukov/webogram#1287]
See https://telegram.org/blog/pin-and-ifttt

**Comment by Ryuno-Ki** [comment]
Any progress? It has a „Ready” label.

@zhukov Could you point to where get information about pinned message in terms of code/Telegram API?

**Comment by stek29** [comment]
@Ryuno-Ki You should handle `updateDialogPinned` and `updatePinnedDialogs`.
You should also check `dialog.pinned` for `messages.getDialogs`. 
And it's better to call `messages.getPinnedDialogs` once, and do all `messages.getDialogs` calls with `exclude_pinned=true`.
And for modifying the pinned list you should use `messages.toggleDialogPin` and `messages.reorderPinnedDialogs`.

**Comment by zhukov** [comment]
@Ryuno-Ki 'ready' label means ready for implementation.
If nobody started working on this, I'll implement it myself.

**Comment by Ryuno-Ki** [comment]
@zhukov Would be handy to give pointers to where to start, so you don't have to code everything alone.
You currently have the advantage of knowing the code base best, but I can imagine it is a huge load looking at all those open issues/requests.

**Comment by stek29** [comment]
@Ryuno-Ki Actually I have a quick'n'dirty implementation of this, but the order gets messed up and I don't have any idea on how to fix it. That's why I stopped working on it and haven't added actual "Pin/Unpin" buttons.

**Comment by Ryuno-Ki** [comment]
Even if it's Quick&Dirty you could publish it so others could have a look at it and comment it.

**Comment by stek29** [comment]
@Ryuno-Ki Sure I will, once I get to PC I had it installed on.

**Comment by Ryuno-Ki** [comment]
@stek29 Any updates?

**Comment by stek29** [comment]
@Ryuno-Ki I haven't even looked on what I just commited, and I wrote in long time ago. I'm not even sure it's working now. 
https://github.com/stek29/webogram/tree/pinned-chats

**Comment by Ryuno-Ki** [comment]
A [comparison](https://github.com/zhukov/webogram/compare/master...stek29:pinned-chats) would be more helpful, I think.

**Comment by azzimuth** [comment]
Is the functionality to pin the conversations implemented? It doesn't look so when I look into the code or try to find the way to pin a chat in Webogram...

**Comment by stek29** [comment]
I think it wasn't deployed yet

**Comment by orschiro** [comment]
Now it's there. So cool! :-)

**Comment by azzimuth** [comment]
Not really. Still no way to pin a chat.

**Comment by orschiro** [comment]
You are right. But at least they show up as pinned now if you had pinned them on your mobile app.

**Comment by stek29** [comment]
If it is still not possible to pin chats in Webogram, then this issue shouldn't marked as fixed.

**Comment by shentino** [comment]
Please explain why it's not possible to pin a chat, but still possible to see pins on chats that were pinned on a different client?

I pin a chat on desktop, and it shows up pinned on web.

**Comment by azzimuth** [comment]
@shentino the functionality to pin chats is not yet implemented in Webogram. Only the functionality to display the pinned chats is implemented atm.

**Comment by hooddy** [comment]
I think there is good to add pin into chat menu (ellipsis in right upper corner).
